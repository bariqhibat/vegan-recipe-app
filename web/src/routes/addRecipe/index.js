import React, { useContext, useEffect, useState } from "react";
import {
  MinusCircleOutlined,
  PlusOutlined,
  LoadingOutlined,
} from "@ant-design/icons";
import ImgCrop from "antd-img-crop";
import {
  Button,
  Spin,
  Form,
  Input,
  message,
  Upload,
  Space,
  Row,
  Col,
  InputNumber,
  Card,
} from "antd";

import AddRecipeContainer from "./AddRecipeContainer";
import AddRecipeContext from "./AddRecipeContext";

const loadingIcon = <LoadingOutlined style={{ fontSize: 24 }} spin />;

const Content = () => {
  const {
    createRecipe,
    recipeUUID,
    recipe,
    updateRecipeById,
    history,
    createRecipeLoading,
    updateRecipeByIdLoading,
    // uploadFile,
  } = useContext(AddRecipeContext);
  const [imageUrl, setImageUrl] = useState(recipe?.image);
  const [giveLoading, setGiveLoading] = useState(false);

  const [formRef] = Form.useForm();

  // ! APOLLO UPLOAD
  // const onDrop = useCallback(
  //   async ([file]) => {
  //     console.log({ file });
  //     const response = await uploadFile({ variables: { file } });

  //     console.log({ response });
  //   },
  //   [uploadFile]
  // );
  // const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop });

  formRef.setFieldsValue({
    ...recipe,
    instructions: recipe?.instructions.map((e) => ({ description: e })),
  });

  useEffect(() => {
    if (imageUrl !== recipe?.image) {
      setImageUrl(recipe?.image);
    }
  }, [recipe]);

  const HOST_URL =
    (process.env.NODE_ENV || "").trim() !== "production"
      ? process.env.REACT_APP_PROD_HTTP_SERVER_URL
      : process.env.REACT_APP_DEV_HTTP_SERVER_URL;

  const handleChange = (info) => {
    if (info.file.status === "uploading") {
      if (!giveLoading) {
        message.loading("Uploading file...");
        setGiveLoading(true);
      }
      return;
    }
    if (info.file.status === "done") {
      // Get this url from response in real world.
      message.success("File uploaded!");
      setGiveLoading(false);
      setImageUrl(info.file.response.data);
    }
  };

  const beforeUpload = (file) => {
    const isJpgOrPng = file.type === "image/jpeg" || file.type === "image/png";
    if (!isJpgOrPng) {
      message.error("You can only upload JPG/PNG file!");
    }
    const isLt2M = file.size / 1024 / 1024 < 2;
    if (!isLt2M) {
      message.error("Image must smaller than 2MB!");
    }
    return isJpgOrPng && isLt2M;
  };

  const onFinish = async (val) => {
    if (!recipeUUID) {
      const response = await createRecipe({
        variables: {
          title: val.title,
          image: imageUrl,
          ingredients: val.ingredients,
          instructions: val.instructions.map((e) => e.description),
        },
      });

      if (response?.data?.createRecipe?.ok) {
        history.push("/myRecipe");
        message.success("Recipe created!");
      }
    } else {
      const response = await updateRecipeById({
        variables: {
          recipeUUID,
          title: val.title,
          image: imageUrl,
          ingredients: val.ingredients.map((e) => ({
            name: e.name,
            amount: e.amount,
          })),
          instructions: val.instructions.map((e) => e.description),
        },
      });

      if (response?.data?.updateRecipeById?.ok) {
        history.push("/myRecipe");
        message.success("Recipe edited!");
      }
    }
  };
  return (
    <>
      <div
        style={{
          paddingTop: "2rem",
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Spin
          indicator={loadingIcon}
          spinning={updateRecipeByIdLoading || createRecipeLoading}
        >
          <Card
            style={{
              padding: "2rem 5rem",
              maxWidth: "50rem",
              minWidth: "40rem",
              width: "100%",
            }}
          >
            <h2>{recipeUUID ? "Change" : "Create"} a recipe</h2>
            <div
              style={{
                textAlign: "center",
                alignContent: "center",
              }}
            >
              <Form onFinish={onFinish} layout="vertical" form={formRef}>
                <Form.Item name="title" label="Title">
                  <Input placeholder="Chinese eggplant with spicy szechuan sauce" />
                </Form.Item>
                <Form.Item label="Image" name="file">
                  <ImgCrop rotate>
                    <Upload
                      name="file"
                      listType="picture-card"
                      className="avatar-uploader"
                      showUploadList={false}
                      action={HOST_URL.concat("/uploads")}
                      beforeUpload={beforeUpload}
                      onChange={handleChange}
                    >
                      {imageUrl ? (
                        <img
                          src={imageUrl}
                          alt="avatar"
                          style={{ width: "100%" }}
                        />
                      ) : (
                        <div>
                          <PlusOutlined />
                          <p>Upload</p>
                        </div>
                      )}
                    </Upload>
                  </ImgCrop>
                </Form.Item>
                <div>
                  <Form.List name="ingredients">
                    {(fields, { add, remove }) => (
                      <>
                        {fields.map((field) => (
                          <Space
                            key={field.key}
                            align="baseline"
                            style={{ marginBottom: "0.8rem" }}
                          >
                            <Row gutter={8}>
                              <Col style={{ width: "40%" }}>
                                <Form.Item
                                  {...field}
                                  name={[field.name, "name"]}
                                  fieldKey={[field.fieldKey, "name"]}
                                  label="Ingredient name"
                                  style={{ margin: "auto" }}
                                  rules={[
                                    {
                                      required: true,
                                      message: "Missing ingredient name",
                                    },
                                  ]}
                                >
                                  <Input
                                    placeholder="Carrot"
                                    style={{
                                      width: "100%",
                                      padding: "0.25rem 0.75rem",
                                      outline: "none",
                                    }}
                                  />
                                </Form.Item>
                              </Col>
                              <Col style={{ width: "40%" }}>
                                <Form.Item
                                  {...field}
                                  name={[field.name, "amount"]}
                                  fieldKey={[field.fieldKey, "amount"]}
                                  label="Ingredient amount"
                                  style={{ margin: "auto" }}
                                  rules={[
                                    {
                                      required: true,
                                      message: "Missing ingredient amount",
                                    },
                                  ]}
                                >
                                  <InputNumber
                                    min={1}
                                    style={{
                                      width: "100%",
                                      outline: "none",
                                    }}
                                  />
                                </Form.Item>
                              </Col>
                              <Col style={{ width: "20%" }}>
                                <MinusCircleOutlined
                                  onClick={() => remove(field.name)}
                                />
                              </Col>
                            </Row>
                          </Space>
                        ))}
                        <Form.Item>
                          <Button
                            type="dashed"
                            onClick={() => add()}
                            block
                            icon={<PlusOutlined />}
                          >
                            Click here to add ingredient
                          </Button>
                        </Form.Item>
                      </>
                    )}
                  </Form.List>
                </div>
                <div>
                  <Form.List name="instructions">
                    {(fields, { add, remove }) => (
                      <>
                        {fields.map((field, idx) => (
                          <Row
                            key={field.key}
                            align="baseline"
                            style={{ marginBottom: "0.8rem" }}
                          >
                            <Col style={{ width: "80%" }}>
                              <Form.Item
                                {...field}
                                label={`Instruction ${idx + 1}`}
                                name={[field.name, "description"]}
                                fieldKey={[field.fieldKey, "description"]}
                                rules={[
                                  {
                                    required: true,
                                    message: "Missing instruction",
                                  },
                                ]}
                              >
                                <Input.TextArea
                                  rows={4}
                                  placeholder="Instruction..."
                                  style={{
                                    width: "100%",
                                    padding: "0.25rem 0.75rem",
                                    outline: "none",
                                  }}
                                />
                              </Form.Item>
                            </Col>
                            <Col style={{ width: "20%" }}>
                              <MinusCircleOutlined
                                onClick={() => remove(field.name)}
                              />
                            </Col>
                          </Row>
                        ))}
                        <Form.Item>
                          <Button
                            type="dashed"
                            onClick={() => add()}
                            block
                            icon={<PlusOutlined />}
                          >
                            Click here to add instruction
                          </Button>
                        </Form.Item>
                      </>
                    )}
                  </Form.List>
                </div>
                <Button htmlType="submit">Submit</Button>
              </Form>
            </div>
          </Card>
        </Spin>
      </div>
    </>
  );
};
export default ({ jestMock }) => {
  if (jestMock) jestMock("addRecipe rendered!");
  return (
    <AddRecipeContainer>
      <Content />
    </AddRecipeContainer>
  );
};
