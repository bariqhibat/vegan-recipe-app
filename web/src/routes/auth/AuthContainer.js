import React from "react";
import { withRouter, Redirect } from "react-router-dom";
import { useMutation, useLazyQuery } from "@apollo/client";
import { EMAIL_SIGNUP, EMAIL_LOGIN, ME_QUERY } from "../../graphql/Users";

import AuthContext from "./AuthContext";

const Container = ({ history, setLoggedIn, ...props }) => {
  let loading = true;
  let error = null;

  const { children } = props;

  // Get params from routes
  // eslint-disable-next-line

  // Run queries

  // GQL mutations
  const [
    createUser,
    { loading: createUserLoading, error: createUserError },
  ] = useMutation(EMAIL_SIGNUP);

  const [login, { loading: loginLoading, error: loginError }] = useMutation(
    EMAIL_LOGIN
  );

  const [
    fetchMe,
    {
      // data: { meQuery: { user } = {} } = {},
      loading: meQueryLoading,
      error: meQueryError,
    },
  ] = useLazyQuery(ME_QUERY);

  try {
    loading = createUserLoading || loginLoading || meQueryLoading;

    // Error view
    error = createUserError || loginError || meQueryError;

    if (error) {
      throw error;
    }

    return (
      <AuthContext.Provider
        value={{
          setLoggedIn,
          history,
          loading,
          // Hooks
          // Data from server
          fetchMe,
          // Mutation
          createUser,
          login,
        }}
      >
        {children}
      </AuthContext.Provider>
    );
    // eslint-disable-next-line
  } catch (error) {
    return (
      <Redirect
        to={{
          pathname: "/500",
          state: { error },
        }}
      />
    );
  }
};

export default withRouter(Container);
