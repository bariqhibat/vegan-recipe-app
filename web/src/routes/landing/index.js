import React, { useContext } from "react";
import { Row, Col, Typography } from "antd";

import RecipeCard from "../../components/RecipeCard";
import PlusCard from "../../components/PlusCard";

import LandingContainer from "./LandingContainer";
import LandingContext from "./LandingContext";

const { Title } = Typography;

const LandingContent = () => {
  const { recipes, history, skeletonLoading } = useContext(LandingContext);
  return (
    <div>
      <div
        style={{
          backgroundColor: "#34495e",
          color: "#fff",
          padding: "10rem",
        }}
      >
        <Row gutter={8} justify="space-between" style={{ height: "20rem" }}>
          <Col lg={12} md={24} xs={24}>
            <Title style={{ color: "#fff" }}>LAVA</Title>
            <Title level={2} style={{ color: "#fff" }}>
              Recipe for everyone
            </Title>
            <h1 style={{ color: "#fff" }}>
              Be a pioneer to enlighten millions of people
            </h1>
            <h1 style={{ color: "#fff" }}>
              Start on adding your recipes <strong>now</strong>!
            </h1>
          </Col>
          <Col lg={12} md={0} xs={0}>
            <img
              src="http://keyword-hero.com/wp-content/uploads/2017/04/Cart-Hero.png"
              alt="hero image"
              style={{
                height: "100%",
                position: "absolute",
                right: 0,
              }}
            />
          </Col>
        </Row>
      </div>
      <div
        style={{
          margin: "auto",
          padding: "0 1rem",
          textAlign: "center",
          paddingTop: "3rem",
        }}
      >
        <Title level={2}>Our current recipes</Title>
        <div style={{ margin: "auto", padding: "1rem" }}>
          <Row>
            {skeletonLoading ? (
              <>
                {[
                  { id: "1" },
                  { id: "2" },
                  { id: "3" },
                  { id: "4" },
                  { id: "5" },
                  { id: "6" },
                ].map(({ id }) => (
                  <Col lg={8} md={12} sm={24} key={id}>
                    <RecipeCard isRecipe={false} />
                  </Col>
                ))}
              </>
            ) : (
              <>
                <Col lg={8} md={12} sm={24} xs={24}>
                  <PlusCard />
                </Col>
                {recipes?.map((recipe) => (
                  <Col lg={8} md={12} sm={24} xs={24} key={recipe._id}>
                    <RecipeCard
                      fromLanding
                      title={recipe.title}
                      image={recipe.image}
                      onClick={() => history.push(`/recipe/${recipe._id}`)}
                    />
                  </Col>
                ))}
              </>
            )}
          </Row>
        </div>
      </div>
    </div>
  );
};

export default ({ jestMock }) => {
  if (jestMock) jestMock("landing rendered!");
  return (
    <LandingContainer>
      <LandingContent />
    </LandingContainer>
  );
};
