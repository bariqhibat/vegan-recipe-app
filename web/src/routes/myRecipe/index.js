import React, { useContext } from "react";
import { Col, Row, Typography } from "antd";

import RecipeCard from "../../components/RecipeCard";
import PlusCard from "../../components/PlusCard";
import MyRecipeContainer from "./MyRecipeContainer";
import MyRecipeContext from "./MyRecipeContext";

const { Title } = Typography;

const capitalizeFirstLetter = (string) => {
  if (!string) {
    return "";
  }
  return string.charAt(0).toUpperCase() + string.slice(1);
};

const Content = ({ initialState }) => {
  const {
    history,
    recipes,
    deleteRecipeById,
    skeletonLoading,
    me,
  } = useContext(MyRecipeContext);
  return (
    <div>
      {initialState?.justLogin && (
        <div
          style={{
            backgroundColor: "#34495e",
            color: "#fff",
            padding: "5rem",
          }}
        >
          <Title style={{ color: "#fff" }}>
            Welcome back, {capitalizeFirstLetter(me?.firstName)}
          </Title>
          <br />
          <br />
          <br />
          <Row gutter={8} justify="space-between">
            <Col lg={12} md={24} xs={24}>
              <Title level={3} style={{ color: "#fff" }}>
                So glad you're back!
                <br />
                This place hasn't been the same without you
              </Title>
            </Col>
            <Col lg={12} md={24} xs={24}>
              <Title level={4} style={{ color: "#fff" }}>
                <i>
                  "One of the marvelous things about community is that it
                  enables us to welcome and help people in a way we couldn't as
                  individuals. When we pool our strength and share the work and
                  responsibility, we can welcome many people, even those in deep
                  distress, and perhaps help them find self-confidence and inner
                  healing."
                </i>
              </Title>
              <Title
                level={5}
                style={{ color: "#fff", position: "absolute", right: 0 }}
              >
                â€• Jean Vanier, Community And Growth
              </Title>
            </Col>
          </Row>
        </div>
      )}
      <div
        style={{
          margin: "auto",
          padding: "1rem",
          textAlign: "center",
          paddingTop: "3rem",
        }}
      >
        <Title level={2}>Your recipes</Title>
        <div style={{ margin: "auto", padding: "1rem" }}>
          <Row>
            {skeletonLoading ? (
              <>
                {[
                  { id: "1" },
                  { id: "2" },
                  { id: "3" },
                  { id: "4" },
                  { id: "5" },
                  { id: "6" },
                ].map(({ id }) => (
                  <Col lg={8} md={12} sm={24} key={id}>
                    <RecipeCard isRecipe={false} />
                  </Col>
                ))}
              </>
            ) : (
              <>
                <Col lg={8} md={12} sm={24} xs={24}>
                  <PlusCard />
                </Col>
                {recipes?.map((recipe) => (
                  <Col lg={8} md={12} sm={24} xs={24} key={recipe._id}>
                    <RecipeCard
                      _id={recipe._id}
                      title={recipe.title}
                      image={recipe.image}
                      onClick={() => history.push(`/recipe/${recipe._id}`)}
                      deleteRecipeById={deleteRecipeById}
                    />
                  </Col>
                ))}
              </>
            )}
          </Row>
        </div>
      </div>
    </div>
  );
};

export default ({ jestMock, location }) => {
  if (jestMock) jestMock("myRecipe rendered!");
  return (
    <MyRecipeContainer>
      <Content initialState={location?.state} />;
    </MyRecipeContainer>
  );
};
