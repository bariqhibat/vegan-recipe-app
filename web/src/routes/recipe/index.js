import React, { useContext } from "react";
import { Card, Row, Col, Spin, List } from "antd";
import RecipeContainer from "./RecipeContainer";
import RecipeContext from "./RecipeContext";

const Content = () => {
  const { recipe, loading } = useContext(RecipeContext);
  const capitalizeFirstLetter = (string) => {
    if (!string) {
      return "";
    }
    return string.charAt(0).toUpperCase() + string.slice(1);
  };

  return (
    <Spin spinning={loading}>
      <Card style={{ margin: "2rem 2rem" }}>
        <Row>
          <Col md={18}>
            <div>
              <h2>{recipe?.title.toUpperCase()}</h2>
              <p>
                <i>
                  Recipe by{" "}
                  <strong>
                    {capitalizeFirstLetter(recipe?.userUUID?.firstName) ||
                      "Anonymous"}
                  </strong>
                </i>
              </p>

              <br />
              <h3>Ingredients</h3>
              <div>
                {recipe?.ingredients.map((e) => (
                  <p>
                    {e.name} - {e.amount}
                  </p>
                ))}
              </div>

              <br />
              <h3>Instructions</h3>

              <div>
                <List
                  bordered
                  dataSource={recipe?.instructions}
                  style={{ width: "50%" }}
                  renderItem={(item) => <List.Item>{item}</List.Item>}
                />
              </div>
            </div>
          </Col>
          {recipe?.image && (
            <Col md={6}>
              <div style={{ textAlign: "right" }}>
                <img
                  style={{ maxWidth: "100%" }}
                  src={
                    recipe?.image ||
                    "https://bakingamoment.com/wp-content/uploads/2018/07/IMG_9113-best-waffle-recipe-square.jpg"
                  }
                  alt="dasd"
                />
                {recipe?.userUUID ? (
                  <p>
                    Photo by{" "}
                    <strong>
                      {capitalizeFirstLetter(recipe?.userUUID?.firstName)}
                    </strong>
                  </p>
                ) : (
                  <p>
                    This recipe is submitted by <strong>NON-USER</strong>
                  </p>
                )}
              </div>
            </Col>
          )}
        </Row>
      </Card>
    </Spin>
  );
};

export default ({ jestMock }) => {
  if (jestMock) jestMock("recipe rendered!");

  return (
    <RecipeContainer>
      <Content />
    </RecipeContainer>
  );
};
