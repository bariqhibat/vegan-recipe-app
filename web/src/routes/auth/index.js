import React from "react";
import { Card } from "antd";

import LoginForm from "./components/LoginForm";
import SignUpForm from "./components/SignupForm";

import AuthContainer from "./AuthContainer";

export const Login = ({ setLoggedIn, jestMock }) => {
  if (jestMock) jestMock("login rendered!");
  return (
    <AuthContainer setLoggedIn={setLoggedIn}>
      <div
        style={{
          paddingTop: "2rem",
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Card
          style={{
            width: "85%",
            maxWidth: "500px",
            height: "fit-content",
            padding: "2rem",
            border: "1px solid white",
          }}
        >
          <LoginForm setLoggedIn={setLoggedIn} />
        </Card>
      </div>
    </AuthContainer>
  );
};

export const SignUp = ({ setLoggedIn, jestMock }) => {
  if (jestMock) jestMock("signUp rendered!");
  return (
    <AuthContainer setLoggedIn={setLoggedIn}>
      <div
        style={{
          paddingTop: "2rem",
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Card
          style={{
            width: "85%",
            maxWidth: "500px",
            height: "fit-content",
            padding: "2rem",
            border: "1px solid white",
          }}
        >
          <SignUpForm />
        </Card>
      </div>
    </AuthContainer>
  );
};
