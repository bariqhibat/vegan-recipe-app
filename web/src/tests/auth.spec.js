import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, waitFor } from "@testing-library/react";
import ApolloMockedProvider from "./utils/provider";

import { Login, SignUp } from "../routes/auth";

const jestMock = jest.fn();

describe("Login", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("Accessability", () => {
    test("Snapshot", async () => {
      const { asFragment } = render(
        <ApolloMockedProvider>
          <Router>
            <Login jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      await waitFor(() => expect(asFragment()).toMatchSnapshot());
    });
    test("Mock", async () => {
      render(
        <ApolloMockedProvider showErrors={false}>
          <Router>
            <Login jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      expect(jestMock).toHaveBeenCalled();
      await waitFor(() =>
        expect(jestMock).toHaveBeenLastCalledWith("login rendered!")
      );
    });
  });
});

describe("Sign Up", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("Accessability", () => {
    test("Snapshot", async () => {
      const { asFragment } = render(
        <ApolloMockedProvider>
          <Router>
            <SignUp jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      await waitFor(() => expect(asFragment()).toMatchSnapshot());
    });
    test("Mock", async () => {
      render(
        <ApolloMockedProvider showErrors={false}>
          <Router>
            <SignUp jestMock={jestMock} />
          </Router>
        </ApolloMockedProvider>
      );

      expect(jestMock).toHaveBeenCalled();
      await waitFor(() =>
        expect(jestMock).toHaveBeenLastCalledWith("signUp rendered!")
      );
    });
  });
});
