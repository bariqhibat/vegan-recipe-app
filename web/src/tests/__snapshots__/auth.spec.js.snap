// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Accessability Snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding-top: 2rem; display: flex; justify-content: center;"
  >
    <div
      class="ant-card ant-card-bordered"
      style="width: 85%; max-width: 500px; padding: 2rem; border: 1px solid white;"
    >
      <div
        class="ant-card-body"
      >
        <div
          style="text-align: center; display: flex; justify-content: center; width: 100%;"
        >
          <div>
            <h1>
              SIGN IN
            </h1>
            <div>
              <form
                class="ant-form ant-form-vertical"
                style="padding: 1rem 0px;"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-24 ant-form-item-label ant-form-item-label-left"
                  >
                    <label
                      class="ant-form-item-required"
                      for="email"
                      title="Email"
                    >
                      Email
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input"
                          id="email"
                          placeholder="john@john.com"
                          style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-col-24 ant-form-item-label ant-form-item-label-left"
                  >
                    <label
                      class="ant-form-item-required"
                      for="password"
                      title="Password"
                    >
                      Password
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-password"
                          style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                        >
                          <input
                            action="click"
                            class="ant-input"
                            id="password"
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              aria-label="eye-invisible"
                              class="anticon anticon-eye-invisible ant-input-password-icon"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="eye-invisible"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                />
                                <path
                                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn"
                          type="submit"
                        >
                          <span>
                            Submit
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <a
              href="/signup"
              style="color: rgb(249, 100, 45);"
            >
              <strong>
                Sign up instead
              </strong>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Sign Up Accessability Snapshot 1`] = `
<DocumentFragment>
  <div
    style="padding-top: 2rem; display: flex; justify-content: center;"
  >
    <div
      class="ant-card ant-card-bordered"
      style="width: 85%; max-width: 500px; padding: 2rem; border: 1px solid white;"
    >
      <div
        class="ant-card-body"
      >
        <div
          style="text-align: center; display: flex; justify-content: center; width: 100%;"
        >
          <div>
            <h1>
              SIGN UP
            </h1>
            <div>
              <div
                class="ant-row ant-row-center ant-row-middle"
                type="flex"
              >
                <form
                  class="ant-form ant-form-vertical"
                  justify="center"
                  style="padding: 1rem 0px; width: 100%;"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="firstName"
                        title="First name"
                      >
                        First name
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input"
                            id="firstName"
                            placeholder="John"
                            style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="lastName"
                        title="Last name"
                      >
                        Last name
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input"
                            id="lastName"
                            placeholder="Doe"
                            style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="email"
                        title="Email"
                      >
                        Email
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input"
                            id="email"
                            placeholder="john@john.com"
                            style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-label"
                    >
                      <label
                        class=""
                        for="password"
                        title="Password"
                      >
                        Password
                      </label>
                    </div>
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <span
                            class="ant-input-affix-wrapper ant-input-password"
                            style="width: 100%; padding: 0.25rem 0.75rem; outline: none;"
                          >
                            <input
                              action="click"
                              class="ant-input"
                              id="password"
                              type="password"
                              value=""
                            />
                            <span
                              class="ant-input-suffix"
                            >
                              <span
                                aria-label="eye-invisible"
                                class="anticon anticon-eye-invisible ant-input-password-icon"
                                role="img"
                                tabindex="-1"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="eye-invisible"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                  />
                                  <path
                                    d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn"
                            type="submit"
                          >
                            <span>
                              Submit
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
            <a
              href="/login"
              style="color: rgb(249, 100, 45);"
            >
              <strong>
                Login instead
              </strong>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
